### PR TITLE
NerdFonts installer script added and consequential change to tab_data TOML file.

### LINKED ISSUE #160: Add option to install Nerd fonts in Linutil
**Is your feature request related to a problem? Please describe.**
Can you add a optional settings to install Nerd fonts in Linutil. When you open it for the first time and if you don't have them preinstalled you get weird symbols and checkboxes instead.

**Describe the solution you'd like**
When you run Linutil there could be additional button that says "Install Fonts" and it could list all optional fonts you could install either as a batch or induvidual. Maybe they could be linked with the official website and that's where you would download them from.
https://www.nerdfonts.com/font-downloads

**Describe alternatives you've considered**
There is a new Flatpak app called Embellish that can install Nerd fonts it could be a option as well 
https://flathub.org/apps/io.github.getnf.embellish

**Additional context**
nerdfonts

### LINKED ISSUE #308: Nerd Fonts Installer Feature Request #160
I am often frustrated when trying to install Nerd Fonts on multiple Linux distributions because it requires manual downloading, unpacking, and moving fonts to the appropriate directories. This process can vary across systems due to different package managers and configurations. Additionally, users who want to install multiple fonts must repeat these steps manually for each one.

I would like to see a Nerd Fonts installer integrated into **linutil** that automates the installation of multiple Nerd Fonts on any Linux distribution. The feature should:

- Automatically detect the user's operating system and use the appropriate package manager to install necessary dependencies like `wget` and `tar`.
- Provide a user-friendly prompt for selecting which Nerd Fonts to install from a comprehensive list.
- Download the selected fonts, extract them to the user's local fonts directory, and refresh the system font cache.
- Be compatible with the existing **linutil** structure, using helper functions from `common-script.sh` for dependency checking and environmental setup.

Alternatives to this automated approach include:

- Manually downloading and installing fonts from the official Nerd Fonts GitHub repository, which can be tedious and error-prone.
- Using external font management tools, but these often do not integrate well with existing systems or are overly complex for users who just want to install a few fonts.

Neither of these alternatives provides the convenience and integration that a dedicated feature within **linutil** would offer.

The script should work seamlessly with different Linux distributions and package managers, leveraging **linutil’s** modular design to ensure compatibility across systems. It should also provide feedback to the user at each step, ensuring transparency during the installation process.

Example output for selecting and installing fonts:

```bash
Select fonts to install (separate with spaces):
---------------------------------------------
 0  -  0xProto Nerd Font
 1  -  3270 Nerd Font
 2  -  Agave Nerd Font
 3  -  AnonymicePro Nerd Font
 4  -  Arimo Nerd Font
...
---------------------------------------------

Enter the numbers of the fonts to install (e.g., '0 1 2'): 1 3 4

Downloading and installing 3270 Nerd Font...
Downloading and installing AnonymicePro Nerd Font...
Downloading and installing Arimo Nerd Font...

Fonts installed and cache updated.

Script:
```sh
#!/bin/sh -e

# Import common utilities
. ./common-script.sh

# Function to prompt user for font selection
prompt_font_selection() {
    fonts=(
        "0xProto Nerd Font"
        "3270 Nerd Font"
        "Agave Nerd Font"
        "AnonymicePro Nerd Font"
        "Arimo Nerd Font"
        "AurulentSansMono Nerd Font"
        "BigBlueTerminal Nerd Font"
        "BitstromWera Nerd Font"
        "BlexMono Nerd Font"
        "CaskaydiaCove Nerd Font"
        "CaskaydiaMono Nerd Font"
        "CodeNewRoman Nerd Font"
        "ComicShannsMono Nerd Font"
        "CommitMono Nerd Font"
        "Cousine Nerd Font"
        "D2Coding Nerd Font"
        "DaddyTimeMono Nerd Font"
        "DejaVuSansMono Nerd Font"
        "DroidSansMono Nerd Font"
        "EnvyCodeR Nerd Font"
        "FantasqueSansMono Nerd Font"
        "FiraCode Nerd Font"
        "FiraMono Nerd Font"
        "GeistMono Nerd Font"
        "GoMono Nerd Font"
        "Gohu Nerd Font"
        "Hack Nerd Font"
        "Hasklug Nerd Font"
        "HeavyDataMono Nerd Font"
        "Hurmit Nerd Font"
        "iM-Writing Nerd Font"
        "Inconsolata Nerd Font"
        "InconsolataGo Nerd Font"
        "Inconsolata LGC Nerd Font"
        "IntoneMono Nerd Font"
        "Iosevka Nerd Font"
        "IosevkaTerm Nerd Font"
        "IosevkaTermSlab Nerd Font"
        "JetBrainsMono Nerd Font"
        "Lekton Nerd Font"
        "Literation Nerd Font"
        "Lilex Nerd Font"
        "MartianMono Nerd Font"
        "Meslo Nerd Font"
        "Monaspice Nerd Font"
        "Monofur Nerd Font"
        "Monoid Nerd Font"
        "Mononoki Nerd Font"
        "M+ Nerd Font"
        "Noto Nerd Font"
        "OpenDyslexic Nerd Font"
        "Overpass Nerd Font"
        "ProFont Nerd Font"
        "ProggyClean Nerd Font"
        "RecMono Nerd Font"
        "RobotoMono Nerd Font"
        "SauceCodePro Nerd Font"
        "ShureTechMono Nerd Font"
        "SpaceMono Nerd Font"
        "Terminess Nerd Font"
        "Tinos Nerd Font"
        "Ubuntu Nerd Font"
        "UbuntuMono Nerd Font"
        "VictorMono Nerd Font"
        "ZedMono Nerd Font"
    )

    echo "Select fonts to install (separate with spaces):"
    echo "---------------------------------------------"
    for i in "${!fonts[@]}"; do
        echo " $i  -  ${fonts[i]}"
    done
    echo "---------------------------------------------"

    read -rp "Enter the numbers of the fonts to install (e.g., '0 1 2'): " font_selection

    echo "Fonts selected: $font_selection"
}

# Function to download and install the selected fonts
download_and_install_fonts() {
    for selection in $font_selection; do
        font=${fonts[$selection]}
        font_name=$(echo "$font" | awk '{print $1}')
        echo "Downloading and installing $font..."
        
        # Check if wget and tar are installed, using common-script.sh helper
        checkCommandRequirements "wget"
        checkCommandRequirements "tar"

        # Download the font
        wget -q --show-progress "https://github.com/ryanoasis/nerd-fonts/releases/latest/download/$font_name.tar.xz" -P "$HOME/tmp"
        
        # Extract and install the font
        mkdir -p ~/.local/share/fonts
        tar -xf "$HOME/tmp/$font_name.tar.xz" -C "$HOME/.local/share/fonts"
        rm "$HOME/tmp/$font_name.tar.xz"
    done

    # Update the font cache
    fc-cache -vf
    echo "Fonts installed and cache updated."
}

# Function to check environment and dependencies
check_environment() {
    echo "Checking environment..."
    checkEnv  # Assuming this is a function from common-script.sh
}

# Main script execution
check_environment
prompt_font_selection
download_and_install_fonts

```

This feature would greatly simplify font management and installation, providing a more streamlined experience for users of linutil.

### PR DESCRIPTION
**Pull request by fam007e**
## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Description
The NerdFonts installer script has been added, allowing users to select and install a variety of NerdFonts through a command-line interface. This feature ensures that the required fonts are downloaded and installed in the user’s `~/.local/share/fonts` directory. Additionally, the font cache is updated to reflect the changes.

A consequential change has been made to the `tab_data` TOML file to integrate the new functionality and ensure smooth operation across the platform.

### Key Features:
- Users can select multiple fonts from a displayed list with scroll functionailty via `less`.
- Fonts are downloaded from the latest NerdFonts GitHub release.
- The font cache is automatically updated after installation.

## Testing
- Ran the script to verify correct downloading and installation of multiple fonts.
- Verified that the font cache updates properly and fonts are available for use.
- Tested the TOML changes for integration with the current configuration.

## Impact
This feature will allow users to easily install and manage NerdFonts, reducing manual download and installation time. There are no new dependencies outside the standard utilities (`curl`, `tar`,`less`) except that of `fc-cache` from `fontconfig`.

## Issue related to PR
- Resolves #160 
- Resolve #308 
- Resolve #309 

## Additional Information
No additional information at this time.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by adamperkowski** [comment]
No way to scroll here when the window with linutil is smaller

image

File: tabs/system-setup/nerdfonts-installer.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                checkCommandRequirements "curl"
+                checkCommandRequirements "tar"
+
+                # Download the font
+                curl -sSLo "$HOME/tmp/$font_name.tar.xz" "https://github.com/ryanoasis/nerd-fonts/releases/latest/download/$font_name.tar.xz"
+
+                # Extract and install the font
+                mkdir -p ~/.local/share/fonts
+                tar -xf "$HOME/tmp/$font_name.tar.xz" -C "$HOME/.local/share/fonts"
+                rm "$HOME/tmp/$font_name.tar.xz"
```
Comment (adamperkowski) [inline tabs/system-setup/nerdfonts-installer.sh:127]:
```suggestion
                curl -sSLo "/tmp/$font_name.tar.xz" "https://github.com/ryanoasis/nerd-fonts/releases/latest/download/$font_name.tar.xz"

                # Extract and install the font
                mkdir -p ~/.local/share/fonts
                tar -xf "/tmp/$font_name.tar.xz" -C "$HOME/.local/share/fonts"
                rm "/tmp/$font_name.tar.xz"
```

File: tabs/system-setup/nerdfonts-installer.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            if [ "$i" = "$selection" ]; then
+                font_name=$(echo "$font" | sed 's/_/ /g')  # Replace underscores with spaces
+                echo "Downloading and installing $font_name..."
+
+                # Check if wget and tar are installed, using common-script.sh helper
+                checkCommandRequirements "curl"
+                checkCommandRequirements "tar"
+
+                # Download the font
+                curl -sSLo "$HOME/tmp/$font_name.tar.xz" "https://github.com/ryanoasis/nerd-fonts/releases/latest/download/$font_name.tar.xz"
```
Comment (adamperkowski) [inline tabs/system-setup/nerdfonts-installer.sh:122]: `curl: (3) URL rejected: Malformed input to a URL function`

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Review by adamperkowski — CHANGES_REQUESTED** [review]
Rename `nerdfonts-installer.sh` to `nerdfonts.sh`

**Comment by adamperkowski** [comment]
The script should also check if `fontconfig` is installed.

File: tabs/system-setup/nerdfonts-installer.sh (hunk 2 above)
Comment (fam007e) [reply tabs/system-setup/nerdfonts-installer.sh:122]: `tar` and `curl` are preloaded so what's the issue here. 
I could not reproduce your error: `curl: (3) URL rejected: Malformed input to a URL function`

**Comment by fam007e** [comment]
> No way to scroll here when the terminal with linutil is smaller
> 
> image

I used my 15" laptop and I also think if it's 14" or 13" its okay so how can you make it `posix` compliant and fit it in within the scope i.e. having to add scroll functionality is beyond my level of understanding... So forgive me I can NOT accomplish that task.

**Comment by adamperkowski** [comment]
> how can you make it posix compliant and fit it in within the scope i.e. having to add scroll functionality is beyond my level of understanding...

You can use `printf` with `read` (#282) or just craft a string and `more` it.

**Comment by fam007e** [comment]
> The script should also check if `fontconfig` is installed.

Shouldn't I add check for `fc-cache` command instead.

File: tabs/system-setup/nerdfonts-installer.sh (hunk 1 above)
Comment (adamperkowski) [reply tabs/system-setup/nerdfonts-installer.sh:127]: @nnyyxxxx I'm pretty sure you're wrong this time. The default `/tmp` permissions are `drwxrwxrwt`. The `t` at the end means that only the owner of a file or the root user can modify.
image

**Comment by adamperkowski** [comment]
> Shouldn't I add check for fc-cache command instead.

You should check if `fontconfig` exists in the system. If not, install it.

You can do `command_exists fc-cache` and if not, install `fontconfig`.

**Comment by fam007e** [comment]
> > Shouldn't I add check for fc-cache command instead.
> 
> You should check if `fontconfig` exists in the system. If not, install it.
> 
> You can do `command_exists fc-cache` and if not, install `fontconfig`.

I think the issue with that is if debian has other utility which installs `fontconfig`.

**Comment by fam007e** [comment]
> > Shouldn't I add check for fc-cache command instead.
> 
> You should check if `fontconfig` exists in the system. If not, install it.
> 
> You can do `command_exists fc-cache` and if not, install `fontconfig`.

May be the "and if not" part be actually implemented in common-script rather than in individual cases like this... 🤔

**Comment by adamperkowski** [comment]
> I think the issue with that is if debian has other utility which installs fontconfig.

You can use 
```sh
case $PACKAGER in
apt|nala)
    # debian stuff
    ;;
...
```

**Comment by adamperkowski** [comment]
@fam007e Draft this and #447 for now. Chris is supposed to be streaming soon.

**Comment by fam007e** [comment]
> @fam007e Draft this and #447 for now. Chris is supposed to be streaming soon.

I had another one of using cli to setup `ssh` to github along with `nerdfonts` and `bootloader-switcher`. These three were made into 1 PR. However, one maintainer suggested to do three separate so I had to use Claude LLM to do the PRs as I am quite busy this week. Sorry about all that. 🙏

**Pull request closed without merging**